thorlcr/master/thmastermain.cpp

```diff
@@ -378,13 +378,26 @@ bool checkClusterRelicateDAFS(IGroup *grp)
     PROGLOG("Checking cluster replicate nodes");
     SocketEndpointArray epa;
     grp->getSocketEndpoints(epa);
-    ForEachItemIn(i1,epa) {
+    SocketEndpointArray epau;
+    ForEachItemIn(i1,epa)
+    {
         epa.element(i1).port = getDaliServixPort();
+        bool add = true;
+        ForEachItemIn(i2,epau)
+        {
+            if (epa.element(i1).equals(epa.element(i2)))
+            {
+                add = false;
+                break;
+            }
+        }
+        if (add)
+            epau.append(epa.element(i1));
```

Comment:
There are a couple of other places that call validateNodes, which would similarly benefit from ensuring they are not performing duplicate checks.
So I think this dudupping code belongs inside validateNodes.